streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -838,24 +867,28 @@ trait ZSink[-R, +E, +A0, -A, +B] { self =>
    * Creates a sink that produces values until one verifies
    * the predicate `f`.
    */
-  final def untilOutput(f: B => Boolean): ZSink[R, E, A0, A, B] =
-    new ZSink[R, E, A0, A, B] {
-      type State = self.State
+  final def untilOutput(f: B => Boolean): ZSink[R, Option[E], A0, A, B] =
```

Comment:
Could you clarify why `Option[E]` is needed?

Reply:
Fixed after discussion in DM.